Update Airflow Release Doc (#23322)
mistercrunch/Airflow: dev/README_RELEASE_AIRFLOW.md

- Check out the 'test' branch

>   For major/minor version release, please follow  the instructions at [[\Optional\] Prepare new release branches and cache](#%5Coptional%5C-prepare-new-release-branches-and-cache) to create the 'test' and 'stable' branches.

Reviewer: ```suggestion
  For major/minor version release, please follow  the instructions at [Prepare new release branches and cache](#%5Coptional%5C-prepare-new-release-branches-and-cache) to create the 'test' and 'stable' branches.
```

Reviewer: We don't need optional in anchor text

    ```shell script
    git checkout v${VERSION_BRANCH}-test
    ```

- Set your version in `setup.py` (without the RC tag)

> - Modify `./scripts/ci/pre_commit/pre_commit_supported_versions.py` and let pre-commit do the job

Reviewer: ```suggestion
- Add supported Airflow version to `./scripts/ci/pre_commit/pre_commit_supported_versions.py` and let pre-commit do the job
```

I have cut Airflow ${VERSION}. This email is calling a vote on the release,

> which will last for 72 hours, from Friday, October 8, 2021 at 4:00 pm UTC

Reviewer 2: ```suggestion
which will last at least 72 hours, from Friday, October 8, 2021 at 4:00 pm UTC
```
